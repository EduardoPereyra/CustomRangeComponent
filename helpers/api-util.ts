export async function getMinMax() {
  const response = await fetch("http://demo0534446.mockable.io/minMax");
  const data = await response.json();
  return data;
}

export async function getPosibleValues() {
  const response = await fetch("http://demo0534446.mockable.io/posibleValues");
  const data = await response.json();
  return data;
}
