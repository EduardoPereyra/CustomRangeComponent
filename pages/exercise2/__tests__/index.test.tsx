import React from "react";
import { render, waitFor } from "@testing-library/react";
import FixedRange from "../index";
import { getPosibleValues } from "../../../helpers/api-util";

jest.mock("../../../helpers/api-util", () => ({
  getPosibleValues: jest.fn(() =>
    Promise.resolve({ rangeValues: [10, 20, 30] })
  ),
}));

test("renders FixedRange component correctly", async () => {
  const { container, debug } = render(<FixedRange />);
  await waitFor(() => expect(getPosibleValues).toHaveBeenCalled());
  expect(container).toMatchSnapshot();
});
