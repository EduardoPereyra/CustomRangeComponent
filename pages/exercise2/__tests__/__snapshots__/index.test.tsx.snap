// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FixedRange component correctly 1`] = `
<div>
  <main
    class="main"
  >
    <a
      href="/"
    >
      X
    </a>
    <div
      class="rangeContainer"
    >
      <h1>
        Fixed Range
      </h1>
      <h2>
        Values: 
        10
         - 
        30
         €
      </h2>
      <div
        aria-valuemax="30"
        aria-valuemin="10"
        aria-valuenow="10"
        class="rangeSlider"
        role="slider"
        style="width: 100%;"
        tabindex="0"
      >
        <div
          class="rangeSliderBar"
          style="width: 0%;"
        >
          <span
            class="rangeSliderHandle"
            style="cursor: grab;"
          >
            <span
              class="rangeSliderLabel false"
            >
              10
              €
            </span>
          </span>
        </div>
        <div
          class="rangeSliderBarBetween"
          style="width: 100%; margin-left: 0%;"
        >
          <span
            class="rangeSliderHandle"
            style="cursor: grab;"
          >
            <span
              class="rangeSliderLabel false"
            >
              30
              €
            </span>
          </span>
        </div>
      </div>
    </div>
  </main>
</div>
`;
